core/domain/suggestion_services.py

```diff
@@ -0,0 +1,367 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Commands that can be used on suggestions."""
+
+from core.domain import exp_services
+from core.domain import feedback_services
+from core.domain import suggestion_domain
+from core.domain import user_services
+from core.platform import models
+import feconf
+
+(feedback_models, suggestion_models) = models.Registry.import_models([
+    models.NAMES.feedback, models.NAMES.suggestion])
+
+DEFAULT_SUGGESTION_THREAD_SUBJECT = 'Suggestion from a learner'
+DEFAULT_SUGGESTION_THREAD_INITIAL_MESSAGE = ''
+
+
+def create_suggestion(
+        suggestion_type, target_type, target_id, target_version_at_submission,
+        author_id, change_cmd, score_category, description,
+        assigned_reviewer_id='', final_reviewer_id=''):
+    """Creates a new SuggestionModel and the corresponding FeedbackThread.
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+        target_type: str. The target entity being edited.
+
+        (The above 2 parameters Should be one of the constants defined in
+        storage/suggestion/gae_models.py.)
+
+        target_id: str. The ID of the target entity being suggested to.
+        target_version_at_submission: int. The version number of the target
+            entity at the time of creation of the suggestion.
+        author_id: str. The ID of the user who submitted the suggestion.
+        change_cmd: dict. The actual content of the suggestion.
+        score_category: str. The category to score the suggestor on.
+        description: str. The description of the changes provided by the author.
+        assigned_reviewer_id: str(optional). The ID of the user assigned to
+            review the suggestion.
+        final_reviewer_id: str(optional). The ID of the reviewer who has
+        accepted/rejected the suggestion.
+    """
+    if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+        thread_id = feedback_services.create_thread(
+            target_id, None, author_id, description,
+            DEFAULT_SUGGESTION_THREAD_SUBJECT)
+
+        # This line will be removed after the feedback thread migration is
+        # complete.
+        thread_id = suggestion_models.TARGET_TYPE_EXPLORATION + '.' + thread_id
+    suggestion_models.SuggestionModel.create(
+        suggestion_type, target_type, target_id, target_version_at_submission,
+        suggestion_models.STATUS_IN_REVIEW, author_id, assigned_reviewer_id,
```

Comment:
Is it actually "in review" if there is no assigned reviewer?

Reply:
Hmmm. I guess not. But till we have the automatic assignment of reviewers (part of milestone 2), I think we need to keep the assigned_reviewer field empty on creation. We can probably add a dropdown for owners/editors to assign one among owners/editors for review. But this will also happen after creation. So for the moment, there may be no assigned reviewer though status is in_review.

Reply:
OK. Is it worth having an additional status type, then (like "received")?

Reply:
Hmmm. Makes sense, I'll add a new status to distinguish between stuff in review and stuff that's not.

Reply:
Sorry, what would "received" indicate?

Reply:
Received would indicate suggestion has been created but no reviewer has been assigned to the suggestion.